main.rb

```diff
@@ -0,0 +1,154 @@
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/CyclomaticComplexity
+require_relative './book'
+require_relative './person'
+require_relative './student'
+require_relative './teacher'
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def all_books
+    if @books.length.positive?
+      @books.each { |book| puts "title: #{book.title}, Author: #{book.author}" }
+    else
+      puts 'There are no books at the moment'
+    end
+  end
+
+  def all_people
+    if @books.length.positive?
+      @people.each do |person|
+        puts "Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    else
+      puts 'There are no people created'
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)?'
+    print 'Select (1) or (2)'
+    print ' '
+    selection = gets.chomp
+
+    case selection
+    when '1'
+      print 'Age: '
+      age = gets.chomp
+
+      print 'Name: '
+      name = gets.chomp
+
+      print 'Has parent permission? [Y/N]: '
+      permission = gets.chomp
+      permission = permission.downcase == 'y'
+
+      @people << Student.new(age, name, permission)
+
+      puts 'student created successfully'
+    when '2'
+      print 'Age: '
+      age = gets.chomp
+
+      print 'Name: '
+      name = gets.chomp
+
+      print 'Specialization: '
+      specialization = gets.chomp
+      @people << Teacher.new(age, specialization, name)
+
+      puts 'Teacher created successfully'
+
+    else
+      puts 'invalid selection'
+
+      nil
+    end
+  end
```

Comment:
- [ ] You need to work on this method `create_person` currently is not working try to go step by step
- first of all you need to ask a user to select if he/she want to create a teacher or a student by choosing between 1 or 2 
- Raise an error  when a user provided a number which is not between 1 and 2
- After selecting whose person he/she want to create you can go ahead with providing all the parameters
and you can always refer to this [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) to see how the app should behave